### ISSUE TITLE
New Decimal type (Mongo 3.4)

### ISSUE THREAD

**Issue by YouriT** [Studio3T/robomongo#1248]
Mongo 3.4 introduced NumberDecimal as a new data type in accordance with BSON RFC about decimal128.

This enables to safely store decimal values. At the moment Robomongo is not able to handle this data type. I suggest an implementation of it as most (all) drivers support it and so does the MongoShell.

More information about this: 
*  http://mongodb.github.io/node-mongodb-native/2.2/tutorials/crud/
*  https://github.com/mongodb/specifications/blob/master/source/bson-decimal128/decimal128.rst
*  https://jira.mongodb.org/browse/DOCS-8736
*  https://jira.mongodb.org/browse/DRIVERS-288

Cheers!

------
**EDIT:**

Examples of Robo behaviour
```javascript
db.getCollection('col').insert({a: NumberDecimal("0.3")})
```
Results in:
```shell
ReferenceError: NumberDecimal is not defined :
@(shell):1:33
```
Expected result:
```javascript
WriteResult({ "nInserted" : 1 })
```
------
If you want to retrieve records:
```javascript
db.getCollection('col').find({})
```
Results in:
```bash
Error: Attempt to use a decimal BSON type when experimental decimal server support is not currently enabled.
```
Expected result:
```javascript
{ "_id" : ObjectId("5841243ef05dc3643d471925"), "a" : NumberDecimal("0.3") }
```

**Comment by simsekgokhan** [comment]
Hi @YouriT, thanks a lot for reporting the problem. As you stated, failure is expected result since Robomongo embeds MongoDB 3.2.

Full support for MongoDB 3.4 and transition to 3.4 drivers will be planned with highest priority. You can follow this ticket https://github.com/paralect/robomongo/issues/1250 or https://twitter.com/Robomongo for updates regarding this feature.

**Comment by simsekgokhan** [comment]
@YouriT 
Please see the beta release includes New Decimal type feature: [Robomongo 1.1 - Beta version with MongoDB 3.4 Support](https://github.com/Studio3T/robomongo/issues/1250#issuecomment-301036770)

**Comment by YouriT** [comment]
Will feedback soon :)

**Comment by juliashibalko** [comment]
Fixed and verified in Robo3T 1.1 version. Closing as fixed.
Please feel free to reopen in any case.
